slowmovie.py

```diff
@@ -10,205 +10,349 @@
-    print('Diplaying frame %d of %s' %(frame,currentVideo))
-    
-    currentPosition = currentPosition + increment 
-    if currentPosition >= frameCount:
-        currentPosition = 0
-        log = open(logdir + '%s<progress'%currentVideo, 'w')
-        log.write(str(currentPosition))
-        log.close() 
-    
-        thisVideo = movieList.index(currentVideo)
-        if thisVideo < len(movieList)-1:
-            currentVideo = movieList[thisVideo+1]
+# ...then try the nowPlaying file, which stores the currently-playing video...
+if not currentVideo and os.path.isfile('nowPlaying'):
+    with open('nowPlaying') as file:
+        lastVideo = file.readline().strip()
+        if os.path.isfile(lastVideo):
+            currentVideo = lastVideo
         else:
-            currentVideo = movieList[0]
+            os.remove('nowPlaying')
+
+# ...then just pick the first video in the videos directory...
+if not currentVideo:
+    # Scan through video directory until you find a video file
+    videos = os.listdir(viddir)
+    for file in videos:
+        if is_vid(file):
+            currentVideo = file
+            break
+
+# ...if none of those worked, exit.
+if not currentVideo:
+    print('No videos found in video directory')
+    sys.exit()
+
+with open('nowPlaying', 'w') as file:
+    file.write(currentVideo)
+
+# Keep track of what video we're playing and which ones are available
+if not args.loop:
+    videos = list(filter(is_vid, os.listdir(viddir)))
+    fileIndex = videos.index(currentVideo)
+
+print(f'Update interval: {args.delay}')
+if not args.random_frames:
+    print(f'Frame increment = {args.increment}')
+
+print(f'With these settings, each minute of 24fps video would take {estimate_runtime(args.delay, args.increment, 60, 24, "d")} to play.')
+print(f'A 120-min movie would last {estimate_runtime(args.delay, args.increment, 120*60, 24, "d")}.')
+
+# Make sure video file passed into CLI is in the videos directory
+if args.file:
+    if args.file in os.listdir(viddir):
+        currentVideo = args.file
+    else:
+        print (f"File '{args.file}' not found")
+
+# Set up EPD
+epd = epd_driver.EPD()
+width = epd.width
+height = epd.height
+
+# Build path to video
+videoFilepath = os.path.join(viddir, currentVideo)
+
+# Build name of log file
+logfile = os.path.join(logdir, currentVideo + '.progress')
+
+# Get the frame and duration infrmation of the video
+videoInfo = get_video_info(videoFilepath)
+
+# Set up the start position based on CLI input or logfiles if either exists
+if not args.random_frames:
+    if args.start:
+        currentFrame = clamp(args.start, 0, videoInfo['frame_count'])
+        print(f'Starting at frame {args.start}')
+    elif os.path.isfile(logfile):
+        # Log files store video progress
+        # Get the stored position for this file from its log
+        with open(logfile) as log:
+            try:
+                currentFrame = clamp(float(log.readline()), 0, videoInfo['frame_count'])
```

Comment:
This may sound radical by why not refactor the log file purpose. There is a new log file for each video - to what purpose. We could store the current frame in ```nowPlaying``` as a dict ```{'file': 'video.mp4', 'frame': 29 }``` and then unify the log into a single ```slowmovie.log``` where this kind of info is displayed on each pass. Using the logging module instead of ```print()``` statements and writing files might be good too. Perhaps too big for this PR - separate PR to overhaul logging?

Reply:
IMO, saving progress works fine the way it is. Of course, the logs folder isn't really for logging, is it? Perhaps it should be renamed.

Reply:
It's more saved data, that's true. As we move more towards a service architecture moving the ```print()``` statements to an actual log file would help with debugging but not really necessary at this stage. It would help file IO (only read one file with both values instead of 2) but I agree it's not the end of the world. We're not trying to fly a rocket or anything here with this code.

Reply:
I'm going to resolve this in the interest of moving this PR along. Overhauling the logging doesn't seem like it fits with this PR. I __do__ think that a unified log file to replace the ```print()``` statements is a good idea; as is storing the current frame information in the ```nowPlaying``` file; however this isn't the time for those kind of changes.

Reply:
I like where this is going, especially spinning it off into a separate enhancement. How do y'all feel about renaming 'log' terminology to 'progress', in preparation for that future work?

This would be renaming the log directory + the terminology in the code. Thoughts?